pkg/manager/manager.go

```diff
@@ -219,53 +223,52 @@ func (m MediaManager) AddMovieToLibrary(ctx context.Context, request AddMovieReq
 	}
 	releases, err := m.SearchIndexers(ctx, indexerIds, categories, *det.Title)
 	if err != nil {
-		log.Debug("failed to search indexer", zap.Any("indexers", indexerIds), zap.Error(err))
+		log.Debugw("failed to search indexer", "indexers", indexerIds, zap.Error(err))
 		return nil, err
 	}
 
 	log.Debugw("releases for consideration", "releases", len(releases))
-	releases = slices.DeleteFunc(releases, rejectReleaseFunc(ctx, det))
+	releases = slices.DeleteFunc(releases, rejectReleaseFunc(ctx, det, profile))
 	log.Debugw("releases after rejection", "releases", len(releases))
 	if len(releases) == 0 {
 		// TODO: This probably isn't really an error... just need to search again later
 		return nil, errors.New("no matching releases found")
 	}
+
 	slices.SortFunc(releases, sortReleaseFunc())
 	chosenRelease := releases[len(releases)-1]
-
-	b, _ := json.Marshal(chosenRelease)
-
-	log.Debug("found release", zap.String("release", string(b)))
-
+	log.Info("found release", "title", chosenRelease.Title)
 	return chosenRelease, nil
 }
 
 // rejectReleaseFunc returns a function that returns true if the given release should be rejected
-func rejectReleaseFunc(ctx context.Context, det *MediaDetails) func(*prowlarr.ReleaseResource) bool {
+func rejectReleaseFunc(ctx context.Context, det *MediaDetails, profile storage.QualityProfile) func(*prowlarr.ReleaseResource) bool {
 	log := logger.FromCtx(ctx)
-	// TODO: Get this dynamically
-	qs := QualitySize{
-		Quality:   "HDTV-720p",
-		Min:       17.1,
-		Preferred: 1999,
-		Max:       2000,
-	}
+
 	return func(r *prowlarr.ReleaseResource) bool {
 		// bytes to megabytes
 		sizeMB := *r.Size >> 20
-		metQuality := MeetsQualitySize(qs, uint64(sizeMB), uint64(*det.Runtime))
-		decision := !metQuality
-		if decision {
-			log.Debugw("rejecting release", "release", r.Title, "metQuality", metQuality, "size", r.Size, "runtime", det.Runtime)
+
+		// items are assumed to be sorted quality so the highest media quality avaiable is selected
+		for _, quality := range profile.Qualities {
+			metQuality := MeetsQualitySize(quality, uint64(sizeMB), uint64(*det.Runtime))
+
+			if metQuality {
+				log.Infow("accepting release", "release", r.Title, "metQuality", metQuality, "size", r.Size, "runtime", det.Runtime)
```

Comment:
Was making this info from debug a conscious choice? I had been thinking it's potentially a lot of output

Reply:
nope, reverted. It does spit out a lot of logs if there are a lot of potential releases pulled back from the tracker